Quiet "Broke affinity for irq" messages during suspend

Quiet the "Broke affinity for irq" messages during suspend. Do this
by conditionally printing the messages depending on the system state,
as originally suggested by Jan when discussing commit e509b8e09c00
("x86: fixup IRQs when CPUs go down during shutdown").

Signed-off-by: Ross Lagerwall <ross.lagerwall@citrix.com>
Suggested-by: Jan Beulich <jbeulich@suse.com>

diff --git a/xen/arch/x86/irq.c b/xen/arch/x86/irq.c
index ff3ac832f4b9..e44e0b394dbb 100644
--- a/xen/arch/x86/irq.c
+++ b/xen/arch/x86/irq.c
@@ -2742,6 +2742,9 @@ void fixup_irqs(void)
 
         spin_unlock(&desc->lock);
 
+	if ( system_state == SYS_STATE_suspend )
+            continue;
+
         if ( !set_affinity )
             printk("Cannot set affinity for IRQ%u\n", irq);
         else if ( break_affinity )
